### PR TITLE
Antismash gff builder: SS review

### PR DESCRIPTION
**Pull request by SantiagoSanchezF**
**Major changes**:
- Changed feature considered a BGC from 'protocluster' to 'region'. Region is the conceptual equivalent of BGC
- Added regions to GFF3.
- CDS now have 'Parent' attrib. to show membership to 'region'.
- Corrected 0-based indexing to 1-based indexing for final GFF3.
- Added header to GFF3.
- Added `cds_tag` argument to script. Improves control the identifier of CDS.
- Save GFF3 to specified file.
**Minor changes**:
- Improved variable naming for code clarity. Also removed some unused vars.
- Changed features attrib object from str to dict. Imprives code clarity.

### CODE REVIEW

File: mgnify_pipelines_toolkit/analysis/assembly/antismash_gff_builder.py [hunk 1]
```diff
@@ -8,121 +23,115 @@
...
-                        score = tool['best_hits'][best_hit]['bitscore']
-                        eval = tool['best_hits'][best_hit]['evalue']
-
-                        if "as_notes" in note_str:
-                            note_str += f'smCOG%3A%20{hit_id}%A{hit_desc}%20%28Score%3A%20{score}%3B%20E-value%3A%20{eval}%29%3B'
-                        else:
-                            note_str += f';as_notes=smCOG%3A%20{hit_id}%A{hit_desc}%20%28Score%3A%20{score}%3B%20E-value%3A%20{eval}%29%3B'
-                        if best_hit in note_dict.keys():
-                            note_dict[best_hit] += note_str
+                if tool['tool'] == 'smcogs':
```
Comment (SantiagoSanchezF) [inline mgnify_pipelines_toolkit/analysis/assembly/antismash_gff_builder.py:115]: Based on my test data, this is redundant with the 'gene_functions' attributes of CDS. But I'm not entirely sure

Comment (chrisAta) [reply mgnify_pipelines_toolkit/analysis/assembly/antismash_gff_builder.py:115]: This gives us the SMCOG ID as well, so it's mostly redundant but there is that one piece of info that makes it worth keeping IMO

**Comment by chrisAta** [comment]
Thanks for the amazing review @SantiagoSanchezF ! This looks much better - I've made some final minor changes and verified them with Tanya who is happy with it. The output now passes the gff3validator tool checks, I had to switch `CDS` to `gene` because `CDS` enforces some rules that requires the phase to be known... if you can have one final look before we merge it that would be great 👍

File: mgnify_pipelines_toolkit/analysis/assembly/antismash_gff_builder.py [hunk 2]
```diff
@@ -119,7 +119,7 @@ def main():
                         score = tool['best_hits'][locus_tag]['bitscore']
                         e_value = tool['best_hits'][locus_tag]['evalue'] # Changed var name to not confuse with built-in func
 
-                        smcog_note = f"smCOG:{hit_id}:{hit_desc.replace(' ', '_')}(Score:{score};E-value:{e_value})"
+                        smcog_note = f"smCOG:{hit_id}:{hit_desc.replace(' ', '_')}(Score:{score}%3BE-value:{e_value})"
```
Comment (SantiagoSanchezF) [inline mgnify_pipelines_toolkit/analysis/assembly/antismash_gff_builder.py:122]: Good catch :)

**Pull request merged**